### PR TITLE
feat: Add Places AutoComplete for Search

### LINKED ISSUE #799: Add Google Place AutoComplete
**Describe the feature you'd like**
For searching events in a specific place, we should use google place autocomplete API.

**Would you like to work on the issue?**
- [X] Yes

### PR DESCRIPTION
**Pull request by theLimitBreaker**
Fixes #799 

Changes: 
1) Add MapBox Autocomplete Place View , as it doesn't need PlayServices 
   So work on both F-droid and PlayStore version :)
   1.1) Changes done in the layout for Better UX
   1.2) Also shows `Recents Location`

2) Change in "Current Location" and change in place of Current Location 
    Because "Current Location button" could not come in between the AutoComplete Fragment
3) Added a custom Round background used in "Current Location Button"
    For better and Nice UX
4) Change in the color of `ic_location_pin` and used `tint` to change the color where needed 

Screenshots for the change:

### CODE REVIEW

**Comment by theLimitBreaker** [comment]
@iamareebjamal  Please Review

File: app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt [hunk 1]
```diff
@@ -57,61 +61,73 @@ class SearchLocationFragment : Fragment() {
...
+        grantResults: IntArray
+    ) {
         when (requestCode) {
             LOCATION_PERMISSION_REQUEST -> {
                 if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                     geoLocationViewModel.configure()
                 } else {
-                    Snackbar.make(rootView, R.string.cannot_fetch_location, Snackbar.LENGTH_SHORT).show()
+                    Snackbar.make(rootView, R.string.cannot_fetch_location, Snackbar.LENGTH_SHORT)
+                        .show()
```
Comment (addiegupta) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: Unnecessary change

Comment (theLimitBreaker) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: I just ran Code Format , it did that automatically.

**Comment by iamareebjamal** [comment]
Great work. Ask your peers to review it first

File: app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt (hunk 1 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: Don't run auto format in PRs.

**Comment by theLimitBreaker** [comment]
> Great work. Ask your peers to review it first

Thank you. And Would take care of this thing first

File: app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt (hunk 1 above)
Comment (theLimitBreaker) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: Do I have to revert this change ?

Comment (ShridharGoel) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: Yes

Comment (theLimitBreaker) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: ok will do it

Comment (theLimitBreaker) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: @ShridharGoel  I have removed unnecessary changes . Please merge this

File: app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt [hunk 2]
```diff
@@ -31,22 +31,26 @@ class SearchLocationFragment : Fragment() {
...
     private val safeArgs: SearchLocationFragmentArgs by navArgs()
 
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+    private val AUTOCOMPLETE_FRAG_TAG = "AutoComplete_Frag"
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: Unnecessary changes.

**Comment by liveHarshit** [comment]
A search bar should be there. Why you remove it? 
Please check search in Eventbrite, how it works.

**Comment by theLimitBreaker** [comment]
> A search bar should be there. Why you remove it?
> Please check search in Eventbrite, how it works.

@liveHarshit there is search bar in the fragment itself . You could see in the gif .

File: app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt (hunk 2 above)
Comment (theLimitBreaker) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: Umm . This is for better view of parameters used.  Still don't want . I would revert it

**Comment by theLimitBreaker** [comment]
> Action bar is missing in search fragment.

@liveHarshit , Autocomplete fragment itself give a back button . Having action bar for back button and fragment also showing backbutton will create UI confusion.

**Comment by theLimitBreaker** [comment]
@liveHarshit eventbrite uses Google's Autocomplete which would require Play services and show the results in custom layout. I could do that . But it won't work for f-droid (because of Play services) .
Current MapBox Autocomplete works on both and provides same UI for both versions

**Comment by liveHarshit** [comment]
> > Action bar is missing in search fragment.
> 
> @liveHarshit , Autocomplete fragment itself give a back button . Having action bar for back button and fragment also showing backbutton will create UI confusion.

I'm asking about the action bar in the search fragment, not in search location fragment.

**Comment by theLimitBreaker** [comment]
> I'm asking about the action bar in the search fragment, not in search location fragment.

I have done no changes in SearchFragment.kt . You could see in `Files Changed`

**Comment by liveHarshit** [comment]
> > I'm asking about the action bar in the search fragment, not in search location fragment.
> 
> I have done no changes in SearchFragment.kt . You could see in `Files Changed`

One change reflects another. Please show() action bar in search fragment inside on create view.

File: app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt [hunk 3]
```diff
@@ -31,22 +31,26 @@ class SearchLocationFragment : Fragment() {
...
         rootView = inflater.inflate(R.layout.fragment_search_location, container, false)
 
         val thisActivity = activity
         if (thisActivity is AppCompatActivity) {
-            thisActivity.supportActionBar?.show()
-            thisActivity.supportActionBar?.title = ""
-            thisActivity.supportActionBar?.setDisplayHomeAsUpEnabled(true)
+            thisActivity.supportActionBar?.hide()
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: Here you hide it in activity, not in the fragment, so you need to show it again in search fragment.

File: app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt [hunk 4]
```diff
@@ -57,29 +61,15 @@ class SearchLocationFragment : Fragment() {
...
-    override fun onResume() {
-        super.onResume()
-        Utils.showSoftKeyboard(context, search)
-    }
-
     private fun checkLocationPermission() {
         val permission =
-            ContextCompat.checkSelfPermission(requireContext(), Manifest.permission.ACCESS_COARSE_LOCATION)
+            ContextCompat.checkSelfPermission(requireContext(),
+                Manifest.permission.ACCESS_COARSE_LOCATION)
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: Unnecessary changes.

File: app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt [hunk 5]
```diff
@@ -91,7 +81,42 @@ class SearchLocationFragment : Fragment() {
         Navigation.findNavController(rootView).popBackStack(fragmentId, false)
     }
 
-    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: Also change override method in MainActivity.

File: app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt (hunk 3 above)
Comment (theLimitBreaker) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: Hmm , that for removing actionbar in SearchLocationFragment.kt. 
Will have to enable the actionbar in onResume() of SearchFragment.kt 
Thanks for pointing it out :) .will do it

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: Not onReasume, show it in onDestroyView of search location fragment.

Comment (theLimitBreaker) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: Yeah sorry . I forgot these are fragments. Will have to call  this in `onDestroyView` .

**Comment by theLimitBreaker** [comment]
@liveHarshit I have updated it Removed unecessary changs and actionbar is also shown

File: app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt [hunk 6]
```diff
@@ -31,22 +31,19 @@ class SearchLocationFragment : Fragment() {
     private val geoLocationViewModel by viewModel<GeoLocationViewModel>()
     private val safeArgs: SearchLocationFragmentArgs by navArgs()
 
+    private val AUTOCOMPLETE_FRAG_TAG = "AutoComplete_Frag"
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: Declare it as constant.

**Comment by liveHarshit** [comment]
> Also change override method in MainActivity.

And what about it?

**Comment by theLimitBreaker** [comment]
> > Also change override method in MainActivity.
> 
> And what about it?

I didn't change Override method then, It was syntax change, that was show, . You could see in files changed

**Comment by liveHarshit** [comment]
Earlier you have removed `onRequestPermissionsResult`. It is calling from MainActivity, so I asked to change it. Now it is okay.

File: app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt (hunk 6 above)
Comment (theLimitBreaker) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: done

**Comment by liveHarshit** [comment]
The current location button at the bottom looks odd. Can you please add it at top of the list like google map.

**Comment by theLimitBreaker** [comment]
>Change in "Current Location" and change in place of Current Location
Because "Current Location button" could not come in between the AutoComplete Fragment

I mentioned this, It could not come in between, search bar and  list.

Alternate way is
- Use Google Autocomplete API,
- get result (JSON)
- Show it in custom view,
 but that also would support only PlayStore only (Due to Play services issue)

F-Droid would still be left

**Comment by liveHarshit** [comment]
@iamareebjamal Your openion?

**Comment by iamareebjamal** [comment]
Seems fine if it is not our UI. It's less code to maintain. @liveHarshit Please approve if everything else is fine

**Comment by iamareebjamal** [comment]
Resolve conflicts

**Comment by theLimitBreaker** [comment]
@iamareebjamal , after merging latest commits. New Voice button is added . But it was added for old layout. Now for adding autoplace frag. I might again have to change the layout . Could I do that ?
